### PR TITLE
Optionally use filepath as part of merge key for merge3

### PR DESCRIPTION
**Pull request by pwittrock**

### CODE REVIEW

File: functions/examples/injection-tshirt-sizes/image/main.go [hunk 1]
```diff
@@ -48,9 +48,9 @@ var cpuSizes = map[string]string{
 
 // memorySizes is the mapping from tshirt-size to memory reservation quantity
 var memorySizes = map[string]string{
-	"small":  "50MiB",
-	"medium": "1GiB",
-	"large":  "32GiB",
+	"small":  "50M",
+	"medium": "1G",
+	"large":  "32G",
```
Comment (droot) [inline functions/examples/injection-tshirt-sizes/image/main.go]: Is this related to this change ?

File: kyaml/kio/filters/merge3.go [hunk 2]
```diff
@@ -110,10 +115,14 @@ func (ts *tuples) add(node *yaml.RNode) error {
...
 	}
+	nodePath := nodeMeta.Annotations[kioutil.PathAnnotation]
 	for i := range ts.list {
 		t := ts.list[i]
+
+		path := t.meta.Annotations[kioutil.PathAnnotation]
 		if t.meta.Name == nodeMeta.Name && t.meta.Namespace == nodeMeta.Namespace &&
-			t.meta.APIVersion == nodeMeta.APIVersion && t.meta.Kind == nodeMeta.Kind {
+			t.meta.APIVersion == nodeMeta.APIVersion && t.meta.Kind == nodeMeta.Kind &&
+			(!ts.mergeOnPath || nodePath == path) {
```
Comment (droot) [inline kyaml/kio/filters/merge3.go]: would be good to add a comment somewhere explaining `why` part of this logic here.

File: functions/examples/injection-tshirt-sizes/image/main.go (hunk 1 above)
Comment (pwittrock) [reply functions/examples/injection-tshirt-sizes/image/main.go]: nope, in a PR that got merged before this

File: kyaml/kio/filters/merge3.go (hunk 2 above)
Comment (pwittrock) [reply kyaml/kio/filters/merge3.go]: PTAL

**Comment by pwittrock** [comment]
PTAL

File: kyaml/kio/filters/merge3.go [hunk 3]
```diff
@@ -112,8 +149,7 @@ func (ts *tuples) add(node *yaml.RNode) error {
 	}
 	for i := range ts.list {
 		t := ts.list[i]
-		if t.meta.Name == nodeMeta.Name && t.meta.Namespace == nodeMeta.Namespace &&
-			t.meta.APIVersion == nodeMeta.APIVersion && t.meta.Kind == nodeMeta.Kind {
+		if ts.isSameResource(t.meta, nodeMeta) {
```
Comment (droot) [inline kyaml/kio/filters/merge3.go]: This makes it very clear. Thanks for the refactor.
